packages/amplify-e2e-tests/src/rds-v2-tests-common/rds-auth-userpool-fields.ts

```diff
@@ -0,0 +1,1925 @@
+        authors: [userName1, userName2],
+        privateContent: 'Private Content updated',
+        ownerContent: 'Owner Content',
+      };
+      const updateResult1 = await user1TodoHelper.update(updateResultSetName, todoUpdated1, setWithOwnerContent);
+      expect(updateResult1.data[updateResultSetName].id).toEqual(todo['id']);
+      expect(updateResult1.data[updateResultSetName].owner).toEqual(userName1);
+      expect(updateResult1.data[updateResultSetName].authors).toEqual([userName1, userName2]);
+      expectNullFields(updateResult1.data[updateResultSetName], ['privateContent', 'ownerContent']);
+
+      // user1 can read the allowed fields
+      const getResult1 = await user1TodoHelper.get(
+        {
+          id: todo['id'],
+        },
+        completeOwnerResultSet,
+        false,
+      );
+      checkOperationResult(getResult1, { ...todo, ...todoUpdated1 }, `get${modelName}`);
+
+      const listTodosResult1 = await user1TodoHelper.list({}, completeOwnerResultSet, `list${modelName}`, false);
+      checkListItemExistence(listTodosResult1, `list${modelName}`, todo['id'], true);
+
+      // unless one has delete access to all fields in the model, delete is expected to fail
+      await expect(
+        async () => await user1TodoHelper.delete(`delete${modelName}`, { id: todo['id'] }, privateResultSet),
+      ).rejects.toThrowErrorMatchingInlineSnapshot(expectedOperationError(`delete${modelName}`, 'Mutation'));
+
+      /*
+      // user2 can update the private field
+      const todoUpdated2 = {
+        id: todo['id'],
+        privateContent: 'Private Content updated 1',
+      };
+      const updateResult2 = await user2TodoHelper.update(updateResultSetName, todoUpdated2, privateResultSet);
+      expect(updateResult2.data[updateResultSetName].id).toEqual(todo['id']);
+      expectNullFields(updateResult2.data[updateResultSetName], ['privateContent']);
+      */
+
+      // user2 can read the allowed fields
+      const user2ReadAllowedSet = `
+        id
+        privateContent
+        ownersContent
+      `;
+      const getResult2 = await user2TodoHelper.get(
+        {
+          id: todo['id'],
+        },
+        user2ReadAllowedSet,
+        false,
+      );
+      checkOperationResult(
+        getResult2,
+        {
+          id: todo['id'],
+          privateContent: todoUpdated1['privateContent'],
+          ownersContent: todo['ownersContent'],
+        },
+        `get${modelName}`,
+      );
+
+      const listTodosResult2 = await user2TodoHelper.list({}, user2ReadAllowedSet, `list${modelName}`, false);
+      checkListItemExistence(listTodosResult2, `list${modelName}`, todo['id'], true);
+
+      // owner can listen to updates on allowed fields
+      const todoRandom = {
+        ...todo,
+        id: Date.now().toString(),
+      };
+      const todoRandomUpdated = {
+        ...todoUpdated1,
+        id: todoRandom.id,
+        owner: userName1,
+        privateContent: 'Private Content updated',
```

Comment:
as above, do we also test subscription updates on `ownersContent`?

Reply:
We test it in `onCreate` since that fields allows only create operation for this model.